modules/ding_webtrekk/ding_webtrekk.module

```diff
@@ -51,24 +46,92 @@ function ding_webtrekk_page_alter(&$page) {
         'option: {}
       };' . $tag_integration_logic,
       'type' => 'inline',
-      'scope' => 'footer',
+      'group' => JS_LIBRARY,
+      'weight' => -21,
     ],
   ];
 
+  // Here we collect and add the general page parameters that aren't dependant
+  // on data and timing from a specific hook implementation.
+  $params = [];
+
+  // Tracks whether the user is logged in and, if logged in, what login provider
+  // was used.
+  if (user_is_logged_in()) {
+    // Note that this separation of logged in users might be problematic if some
+    // pages are also cached for authenticated users.
+    if (module_exists('ding_gatewayf') && _ding_gatewayf_is_logged_in_with_gatewayf()) {
+      $params['p_isloggedin'] = 'NemId';
+
+      // This is the landing page for successful user registration with gatewayf.
+      // If the registration was successful, the users will be automatically
+      // logged in. So by also requiring a logged in user before setting parameter
+      // we reduce the chances of false positives. For example, it will prevent
+      // random access to the URL from anonymous users, from being tracked.
+      if (strpos(current_path(), DING_GATEWAYF_REGISTRATION_SUCCESS_URL) === 0) {
+        $params['p_usercreate'] = 'NemId';
+      }
+    }
+    // Fallback to cpr+pin if none the modules we know about was used to log in.
+    else {
+      $params['p_isloggedin'] = 'cpr+pinkode';
```

Comment:
Note that this will also track Drupal users as logged in with CPR+Pin. I do not know if that is intentional?

Reply:
Good point. I will ask the business about this.

Reply:
I asked and there was no problem with this. They are working on a way to exclude administrative users anyway.